cpp/test/mg/pca.cu

```diff
@@ -0,0 +1,214 @@
+
+    mu = (T*)allocator->allocate(prmsPCA.n_cols * sizeof(T), stream);
+
+    noise_vars =
+      (T*)allocator->allocate(prmsPCA.n_components * sizeof(T), stream);
+
+    ML::PCA::opg::fit(*handle, inParts, desc, components, explained_var,
+                      explained_var_ratio, singular_vals, mu, noise_vars,
+                      prmsPCA, false);
+
+    Matrix::opg::deallocate(h, inParts, desc, myRank, stream);
+  }
+
+  void TearDown() {
+    const std::shared_ptr<deviceAllocator> allocator =
+      handle->getDeviceAllocator();
+    cudaStream_t stream = handle->getStream();
+
+    allocator->deallocate(
+      components, prmsPCA.n_components * prmsPCA.n_cols * sizeof(T), stream);
+
+    allocator->deallocate(explained_var, prmsPCA.n_components * sizeof(T),
+                          stream);
+
+    allocator->deallocate(explained_var_ratio, prmsPCA.n_components * sizeof(T),
+                          stream);
+
+    allocator->deallocate(singular_vals, prmsPCA.n_components * sizeof(T),
+                          stream);
+
+    allocator->deallocate(mu, prmsPCA.n_cols * sizeof(T), stream);
+
+    allocator->deallocate(noise_vars, prmsPCA.n_components * sizeof(T), stream);
+
+    delete handle;
+  }
+
+ protected:
+  PCAOpgParams params;
+  ML::cumlHandle* handle;
+  cudaStream_t stream;
+  int myRank;
+  int totalRanks;
+  ML::paramsPCAMG prmsPCA;
+  T *components, *explained_var, *explained_var_ratio, *singular_vals, *mu,
+    *noise_vars;
+};
+
+const std::vector<PCAOpgParams> inputs = {{20,
+                                           4,
+                                           2,
+                                           ML::mg_solver::COV_EIG_JACOBI,
+                                           {11, 9},
+                                           {1, 0},
+                                           Matrix::LayoutColMajor,
+                                           223548ULL},
+                                          {20,
+                                           4,
+                                           2,
+                                           ML::mg_solver::COV_EIG_DQ,
+                                           {11, 9},
+                                           {1, 0},
+                                           Matrix::LayoutColMajor,
+                                           223548ULL},
+                                          {20,
+                                           4,
+                                           2,
+                                           ML::mg_solver::QR,
+                                           {11, 9},
+                                           {1, 0},
+                                           Matrix::LayoutColMajor,
+                                           223548ULL}};
+
+typedef PCAOpgTest<float> PCAOpgTestF;
+
+TEST_P(PCAOpgTestF, Result) {
+  if (myRank == 0) {
+    MLCommon::myPrintDevVector("Singular Values: ", singular_vals,
```

Comment:
We should the `CUML_LOG_DEBUG` here, and maybe use the `arr2Str` utility function to create a string from a device array.